.github/workflows/build_and_publish_release.yml

```diff
@@ -0,0 +1,66 @@
+name: 📦 Build and Publish a WBS Image Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      image_name:
+        description: 'Image to release'
+        type: choice
+        required: true
+        default: 'wikibase'
+        options:
+          - elasticsearch
+          - quickstatements
+          - wdqs
+          - wdqs-frontend
+          - wdqs-proxy
+          - wikibase
+          - All images with unreleased changes
+      dry_run:
+        description: "Only log what will be done, don't do it yet."
+        type: boolean
+        required: true
+        default: true
+
+permissions:
+  contents: write
+  packages: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: ./.github/actions/setup-environment
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: docker/login-action@v3
+        with:
+          # implicitly docker hub
+          username: wmdetravisbot
+          password: ${{ secrets.WBS_PUBLISH_TOKEN }}
+
+      - name: Git tag release and publish version tagged images to Dockerhub
+        run: |
+          set -e # abort on error
+          set -x # show commands
+
+          git config --global user.name 'wikibase suite github actions bot'
+          git config --global user.email 'wikibase-suite-github-actions-bot@users.noreply.github.com'
+
+          if [ "${{ inputs.dry_run }}" == "true" ]; then
+            DRY_RUN_FLAG="--dry-run"
+          else
+            DRY_RUN_FLAG=""
+          fi
+
+          if [ "${{ inputs.image_name }}" == "All images with unreleased changes" ]; then
+            PROJECT_ARG=""
+          else
+            PROJECT_ARG="-p ${{ inputs.image_name }}"
+          fi
+
+          pnpm exec nx release publish $PROJECT_ARG $DRY_RUN_FLAG
```

Comment:
When this is doing an actual "release: publish to dockerhub" and it triggered manually, where does the git tagging happen then?

Reply:
For now the workflow we discussed has a developer running `./nx release` locally before running this actual publishing step manually on github. Happy to change this to an automatic trigger based upon git version tag matching, but for now the actual kicking off of the release itself (updating versions, changelogs, creating git tags, and pushing) is kicked off locally so we can check it all over and amend the changelog and such as we wish which would be awkward to facilitate in an Actions workflow. 

When I look at it again next week perhaps I'll change the Dockerhub publish action to trigger off the version tags... Do you remember why we decided against that when we discussed on Monday?

Reply:
> Do you remember why we decided against that when we discussed on Monday?

I do not remember deciding against that.

I remember this two workflows:
1. Manually triggered github actions workflow to create the release, git tag it, generate changelogs, etc
2. Git tag triggered github actions workflow running build, docker tag, publish to dockerhub on a release git tag

This way we can run step 1 fully in CI for the default case and need no setup on dev  machines, but can also run it locally when we want customization (such as changelog tuning).

Reply:
Certainly that be added by simply changing the current workflow which does `./nx publish <imageName>` to do `./nx release -p <imageName>`, and then creating another workflow which triggers on the version tag/s to then build and push to Dockerhub. That would in fact give the option to run that command as a Workflow,
or locally, which sounds fine to me. I'll do that, and prefer at least that the Dockerhub publishing is triggered automatically in Actions. 

To be clear there is no "setup" required to run this locally though (just Docker), and the result will be the same.